src/login/components/LoginFormLedger/LoginFormLedger.js

```diff
@@ -56,7 +57,8 @@ export default class LoginFormLedger extends React.Component {
   }
 
   renderActions = () => {
-    const disabled = this.props.disabled || !this.props.deviceInfo;
+    const { publicKey, deviceInfo, progress } = this.props;
+    const disabled = !(publicKey && deviceInfo && (progress !== FAILED));
```

Comment:
I think we can simplify this just by looking at `progress` and `disabled` alone:

```js
const disabled = this.props.disabled || this.props.progress !== LOADED;
```